overlord/configstate/configcore/services.go

```diff
@@ -90,6 +91,72 @@ func switchDisableSSHService(sysd systemd.Systemd, serviceName, value string, op
 	}
 }
 
+// switchDisableConsoleConfService handles the special case of disabling/enabling
+// console-conf on core devices.
+//
+// The command sequence that works to start/stop console-conf after setting
+// the marker file in /var/lib/console-conf/complete is:
+//
+//     systemctl restart 'getty@*' --all
+//     systemctl restart 'serial-getty@*' --all
+//     systemctl restart 'serial-console-conf@*' --all
+//     systemctl restart 'console-conf@*' --all
+//
+// This restart all active getty and console-conf instances, even ones that were
+// started on-demand (eg. on tty2)
+func switchDisableConsoleConfService(sysd systemd.Systemd, serviceName, value string, opts *fsOnlyContext) error {
+	rootDir := dirs.GlobalRootDir
+	if opts != nil {
+		rootDir = opts.RootDir
+	}
+	if err := os.MkdirAll(filepath.Join(rootDir, "/var/lib/console-conf"), 0755); err != nil {
+		return err
+	}
+
+	consoleConfCanary := filepath.Join(rootDir, "/var/lib/console-conf/complete")
+
+	restartServicesOnTTYs := func() error {
+		// getty@ and console-conf@ are template services, that only
+		// exist when an instance is active, typically in a UC20 image
+		// only getty@tty1 is defined as a side effect of being 'wanted'
+		// by the getty.target;
+		// restarting all console-conf@* units ensures on-demand units
+		// started on other ttys are affected too
+		if err := sysd.RestartAll("getty@*"); err != nil {
+			return err
+		}
+		if err := sysd.RestartAll("serial-getty@*"); err != nil {
+			return err
+		}
+		if err := sysd.RestartAll("serial-console-conf@*"); err != nil {
+			return err
+		}
+		return sysd.RestartAll("console-conf@*")
+	}
+
+	switch value {
+	case "true":
+		if err := ioutil.WriteFile(consoleConfCanary, []byte("console-conf has been disabled by snapd system configuration\n"), 0644); err != nil {
+			return err
+		}
+		if opts == nil {
+			return restartServicesOnTTYs()
+		}
+		return nil
+	case "false":
+		err := os.Remove(consoleConfCanary)
+		if err != nil && !os.IsNotExist(err) {
+			return err
+		}
+		if opts == nil {
+			return restartServicesOnTTYs()
+		}
+		return nil
+	default:
+		return fmt.Errorf("option %q has invalid value %q", serviceName, value)
```

Comment:
No for this PR, but I think we have a small issue with this error message also in existing code, we print systemd's service name which is not strictly speaking the option name that user entered, may be confusing.